picard/coverart/providers/caa.py

```diff
@@ -148,7 +191,18 @@ def __init__(self, parent=None):
         self.ui.select_caa_types.clicked.connect(self.select_caa_types)
 
     def load(self):
-        self.ui.cb_image_size.setCurrentIndex(config.setting["caa_image_size"])
+        self.ui.cb_image_size.clear()
+        for item_id, item in _CAA_THUMBNAIL_SIZE_MAP.items():
+            self.ui.cb_image_size.addItem(_(item.label), userData=item_id)
+
+        size = config.setting["caa_image_size"]
+        if size in _CAA_SIZE_COMPAT:
```

Comment:
Rather than introducing this, wouldn't it be better to just have a plugin upgrade hook?

Reply:
It was to avoid an upgrade hook. I pondered both approaches, and this one is much simpler and reliable.

Reply:
I just feel that this more of a compat issue with picard versions and should be kept in upgrade hooks, it makes it easier to track config errors. This approach adds unrelated code to the view IMHO.

Also, why are plugin hooks unreliable or this approach more reliable?

Reply:
Mostly because upgrade hooks don't work well when someone switch between versions, this code would handle a switch, or even a buggy entry.
This is perfectly possible to do with an upgrade hook though, feel free to propose a patch.